src/framework/JetScapeWriterFinalStateStream.cc

```diff
@@ -101,9 +103,16 @@ template <class T> void JetScapeWriterFinalStateStream<T>::WriteEvent() {
       << "\t" << "weight\t" << std::setprecision(15) << GetHeader().GetEventWeight() << std::setprecision(6)
       << "\t" << "EPangle\t" << (GetHeader().GetEventPlaneAngle() > -999 ? GetHeader().GetEventPlaneAngle() : 0)
       << "\t" << "N_" << GetName() << "\t" << particles.size()
-      << centrality_text
+      << centrality_text;
+  if (headerVersion == 3) {
+    output_file
+        << "\t" << "vertex_x\t" << GetHeader().GetVertexX()
+        << "\t" << "vertex_y\t" << GetHeader().GetVertexY()
+        << "\t" << "vertex_z\t" << GetHeader().GetVertexZ();
+  }
+  output_file
       << pt_hat_text
```

Comment:
```suggestion
  << "\t" << "N_" << GetName() << "\t" << particles.size();
  if (headerVersion == 3) {
    output_file
        << "\t" << "vertex_x\t" << GetHeader().GetVertexX()
        << "\t" << "vertex_y\t" << GetHeader().GetVertexY()
        << "\t" << "vertex_z\t" << GetHeader().GetVertexZ();
  }
  output_file
      << centrality_text
      << pt_hat_text;
```

Just to group all of the fully optional options at the end